sphinx_gallery/tests/test_gen_rst.py

```diff
@@ -103,6 +103,26 @@ def test_direct_comment_after_docstring():
     assert result == expected_result
 
 
+def test_final_rst_last_word():
+    """Tests last word in final rst block included as text"""
+    with tempfile.NamedTemporaryFile('w', delete=False) as f:
+        f.write('\n'.join(['"Docstring"',
+                           '# comment only code block',
+                           '#%%',
+                           '# Include this whole sentence.']))
+    try:
+        file_conf, result = sg.split_code_and_text_blocks(f.name)
+    finally:
+        os.remove(f.name)
```

Comment:
The modern way would be:
```
def test_final_rst_last_word(tmpdir):
    fname = str(tmpdir.join('whatever.py'))
    with open(fname, 'w') as fid:
        ...
    ...
```
`tmpdir` is a pytest fixture that will auto-delete when the run is over, makes all this boilerplate `tempfile` stuff unnecessary

Reply:
Thanks! Done.